config/seata-config-core/src/main/java/org/apache/seata/config/ConfigurationFactory.java

```diff
@@ -176,6 +178,67 @@ private static Configuration buildConfiguration() {
+        io.seata.config.Configuration oldConfiguration = new io.seata.config.FileConfiguration(configuration);
+        try {
+            io.seata.config.Configuration configurationSPIInstance = EnhancedServiceLoader.load(
+                io.seata.config.ExtConfigurationProvider.class).provide(oldConfiguration);
+            if (null != configurationSPIInstance) {
+                Configuration configurationSPIInstanceProxy = (Configuration)Proxy.newProxyInstance(
+                    ConfigurationFactory.class.getClassLoader(), new Class[] {Configuration.class},
+                    new OldConfigurationInvocationHandler(configurationSPIInstance));
+                return configurationSPIInstanceProxy;
+            }
+        } catch (EnhancedServiceNotFoundException ignore) {
+            //ignore
+        } catch (Exception exx) {
+            LOGGER.error("failed to load spring configuration :{}", exx.getMessage(), exx);
+        }
+        if (null != configuration) {
+            try {
+                Configuration extConfiguration = EnhancedServiceLoader.load(ExtConfigurationProvider.class, false).provide(
+                    configuration);
+                if (null != extConfiguration) {
+                    return extConfiguration;
+                }
+            } catch (EnhancedServiceNotFoundException ignore) {
+                //ignore
+
+            } catch (Exception exx) {
+                LOGGER.error("failed to load spring configuration :{}", exx.getMessage(), exx);
+            }
+        }
+        return null;
+    }
+
+    private static Configuration getNonSpringConfiguration(String configTypeName) {
+        try {
+            io.seata.config.Configuration oldConfiguration = EnhancedServiceLoader.load(
+                io.seata.config.ConfigurationProvider.class, Objects.requireNonNull(configTypeName)).provide();
+            if (null != oldConfiguration) {
+                Configuration configurationSPIInstanceProxy = (Configuration)Proxy.newProxyInstance(
+                    ConfigurationFactory.class.getClassLoader(), new Class[] {Configuration.class},
+                    new OldConfigurationInvocationHandler(oldConfiguration));
+                return configurationSPIInstanceProxy;
+            }
+        } catch (EnhancedServiceNotFoundException ignore) {
+            //ignore
+        } catch (Exception exx) {
+            LOGGER.error("failed to load non-spring configuration :{}", exx.getMessage(), exx);
+        }
+        try {
+            Configuration configuration = EnhancedServiceLoader.load(ConfigurationProvider.class,
+                Objects.requireNonNull(configTypeName), false).provide();
+            return configuration;
+        } catch (EnhancedServiceNotFoundException ignore) {
+            //ignore
+        } catch (Exception exx) {
+            LOGGER.error("failed to load non-spring configuration :{}", exx.getMessage(), exx);
+        }
+        return ORIGIN_FILE_INSTANCE;
```

Comment:
这里不应该返回ORIGIN_FILE_INSTANCE，更应该直接抛出异常提示，可以比较明确的判断问题
Here, instead of returning ORIGIN_FILE_INSTANCE, it would be more appropriate to directly throw an exception with a clear error message, allowing for a more straightforward identification of the issue.

Reply:
fixed

Reply:
An exception cannot be thrown here because the file mode does not go through SPI. I've broken down the previous logic a bit and extracted the assignment of default values to the outer logic, but overall, it remains consistent with what was there before.